### PR TITLE
NuGetSupport: Improve the message of an OrtIssue

### PR DESCRIPTION
**Pull request by fviernau**
Make it obvious that version ranges and wildcards are not supported, to potentially safe some debugging time.

See also [1], [2].

[1] https://learn.microsoft.com/en-us/nuget/concepts/package-versioning 
[2] https://github.com/oss-review-toolkit/ort/issues/6183

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt [hunk 1]
```diff
@@ -382,3 +387,6 @@ interface XmlPackageFileReader {
      */
     fun getDependencies(definitionFile: File): Set<NuGetDependency>
 }
+
+private fun String.isFixedVersion() =
+    listOf('(', ')', '[', ']', '*', '+').none { it in this } && isNotBlank()
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:392]: How about reusing `VERSION_RANGE_CHARS` (after adding `*` and `+` to it)? Also, maybe make the cheap `isNotBlank()` condition go first.

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt [hunk 2]
```diff
@@ -137,11 +137,16 @@ class NuGetSupport(
...
             if (packageData == null) {
+                val hint = " Support for non-fixed versions, like '${id.version}', is not implemented".takeIf {
+                    !id.version.isFixedVersion()
+                }.orEmpty()
+
                 issues += createAndLogIssue(
                     source = "NuGet",
                     message = "Failed to get package data for '${id.toCoordinates()}' from any of " +
-                            "$registrationsBaseUrls."
+                            "$registrationsBaseUrls. $hint"
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:147]: The separating single space is already part of `hint` and thus should be omitted here.

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt [hunk 3]
```diff
@@ -137,11 +137,16 @@ class NuGetSupport(
             val packageData = registrationsBaseUrls.firstNotNullOfOrNull { getPackageData(id, it) }
 
             if (packageData == null) {
+                val hint = " Support for non-fixed versions, like '${id.version}', is not implemented".takeIf {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:140]: "like" sounds like an example, but in fact you're checking exactly that version, so I'd simply say " Support for non-fixed version '${id.version}' is not implemented." (also note the added trailing dot at the end of the sentence).

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt (hunk 1 above)
Comment (fviernau) [reply analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:392]: Do you mean I should change the value of VERSION_RANGE_CHARS and thereby also the behavior of `Dependency.getId()` ?

Comment (sschuberth) [reply analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:392]: > Do you mean I should change the value of VERSION_RANGE_CHARS and thereby also the behavior of `Dependency.getId()` ?

Yes. The behavior change should not matter, but helps you to reuse the variable. This, however, should be stated in the commit message then.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**